src/main/java/org/opentripplanner/middleware/controllers/api/UserController.java

```diff
@@ -49,4 +80,26 @@ boolean preDeleteHook(OtpUser user, Request req) {
         }
         return true;
     }
+
+    /**
+     * Holds result and message from getUserFromProfile.
+     */
+    static class UserFromProfileResult {
+        public OtpUser user;
+        public String message;
+    }
+
+    /**
+     * HTTP endpoint to get the {@link OtpUser} entity from a {@link Auth0UserProfile}.
+     * (Reminder: for endpoints under 'secure', we add a {@link Auth0UserProfile} to request attributes.)
+     */
+    private OtpUser getUserFromRequest(Request req, Response res) {
+        Auth0UserProfile profile = req.attribute("user");
```

Comment:
This should be replaced with Auth0Connection.getUserFromRequest(Request req);

Reply:
Good suggestion.